### PR TITLE
feat: make quick format bar compatible with code block, update ui

### PR DESCRIPTION
**Pull request by thorseraq**

### CODE REVIEW

File: packages/blocks/src/components/format-quick-bar/format-bar-node.ts [hunk 1]
```diff
@@ -119,14 +127,14 @@ export class FormatQuickBar extends LitElement {
       flexDirection:
         this.showParagraphPanel === 'bottom' ? 'column' : 'column-reverse',
     });
-    return html`<div
+    return html` <div
```
Comment (lawvs) [inline packages/blocks/src/components/format-quick-bar/format-bar-node.ts:130]:
```suggestion
    return html`<div
```

File: packages/blocks/src/components/format-quick-bar/format-bar-node.ts [hunk 2]
```diff
@@ -119,14 +127,14 @@ export class FormatQuickBar extends LitElement {
...
       class="paragraph-panel"
       style="${styles}"
       @mouseover=${this.onHover}
       @mouseout=${this.onHoverEnd}
     >
       ${paragraphButtons.map(
-        ({ flavour, type, name, icon }) => html`<format-bar-button
+        ({ flavour, type, name, icon }) => html` <format-bar-button
```
Comment (lawvs) [inline packages/blocks/src/components/format-quick-bar/format-bar-node.ts:137]:
```suggestion
        ({ flavour, type, name, icon }) => html`<format-bar-button
```

File: packages/blocks/src/components/format-quick-bar/format-bar-node.ts [hunk 3]
```diff
@@ -167,13 +175,14 @@ export class FormatQuickBar extends LitElement {
     const paragraphIcon =
       paragraphButtons.find(btn => btn.type === this.paragraphType)?.icon ??
       paragraphButtons[0].icon;
-    const paragraphItems = html`<format-bar-button
+    const paragraphItems = html` <format-bar-button
```
Comment (lawvs) [inline packages/blocks/src/components/format-quick-bar/format-bar-node.ts:178]:
```suggestion
    const paragraphItems = html`<format-bar-button
```

File: packages/blocks/src/components/format-quick-bar/format-bar-node.ts [hunk 4]
```diff
@@ -182,7 +191,7 @@ export class FormatQuickBar extends LitElement {
       ${formatButtons
         .filter(({ showWhen = () => true }) => showWhen(this.models))
         .map(
-          ({ id, name, icon, action, activeWhen }) => html`<format-bar-button
+          ({ id, name, icon, action, activeWhen }) => html` <format-bar-button
```
Comment (lawvs) [inline packages/blocks/src/components/format-quick-bar/format-bar-node.ts:194]:
```suggestion
          ({ id, name, icon, action, activeWhen }) => html`<format-bar-button
```

File: packages/blocks/src/components/format-quick-bar/icons.ts [hunk 5]
```diff
@@ -288,3 +288,10 @@ export const ArrowDownIcon = svg`<svg
     d="M9.97678 11.318C9.96662 11.3111 9.94678 11.2967 9.9152 11.2699C9.82898 11.1967 9.71943 11.0881 9.52859 10.8972L6.39901 7.76763C6.08659 7.45521 5.58006 7.45521 5.26764 7.76763C4.95522 8.08005 4.95522 8.58658 5.26764 8.899L8.39722 12.0286C8.40353 12.0349 8.40982 12.0412 8.41611 12.0475C8.5813 12.2127 8.73751 12.3689 8.87972 12.4897C9.03447 12.621 9.22442 12.7566 9.46951 12.8363C9.81429 12.9483 10.1857 12.9483 10.5305 12.8363C10.7756 12.7566 10.9655 12.621 11.1203 12.4897C11.2625 12.3689 11.4187 12.2127 11.5839 12.0475L14.7323 8.899C15.0448 8.58658 15.0448 8.08005 14.7323 7.76763C14.4199 7.45521 13.9134 7.45521 13.601 7.76763L10.4714 10.8972C10.2806 11.0881 10.171 11.1967 10.0848 11.2699C10.0532 11.2967 10.0334 11.3111 10.0232 11.318C10.0079 11.3211 9.9921 11.3211 9.97678 11.318Z"
   />
 </svg>`;
+
+export const CodeBlockIcon = svg`
```
Comment (lawvs) [inline packages/blocks/src/components/format-quick-bar/icons.ts:292]: The code icon already exists in icons.ts

https://github.com/toeverything/blocksuite/blob/14e22a29412f583d0a6e2582f22b91d8df13897f/packages/blocks/src/components/format-quick-bar/icons.ts#L135-L151

File: packages/playground/vite.config.ts.timestamp-1671762794292.mjs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+// vite.config.ts
```
Comment (lawvs) [inline packages/playground/vite.config.ts.timestamp-1671762794292.mjs:1]: Please remove this file

File: packages/blocks/src/components/format-quick-bar/format-bar-node.ts [hunk 7]
```diff
@@ -167,13 +175,14 @@ export class FormatQuickBar extends LitElement {
...
       paragraphButtons[0].icon;
-    const paragraphItems = html`<format-bar-button
+    const paragraphItems = html` <format-bar-button
       class="paragraph-button"
       width="52px"
       @mouseover=${this.onHover}
       @mouseout=${this.onHoverEnd}
     >
-      ${paragraphIcon} ${ArrowDownIcon}
+      ${this.isCodeBlockSelection ? CodeBlockIcon : paragraphIcon}
```
Comment (lawvs) [inline packages/blocks/src/components/format-quick-bar/format-bar-node.ts:184]: Please update the icon on `format-quick-bar/config.ts`

**Review by lawvs — CHANGES_REQUESTED** [review]

File: packages/playground/vite.config.ts.timestamp-1671762794292.mjs (hunk 6 above)
Comment (thorseraq) [reply packages/playground/vite.config.ts.timestamp-1671762794292.mjs:1]: Thanks for patience!

File: packages/blocks/src/page-block/utils/container-operations.ts [hunk 8]
```diff
@@ -74,6 +75,28 @@ export function deleteModels(page: Page, models: BaseBlockModel[]) {
...
+  blockProps: Record<string, unknown>
+) {
+  const parent = page.getParent(models[0]);
+  assertExists(parent);
+  const index = parent.children.indexOf(models[0]);
+  const id = page.addBlock(blockProps, parent, index);
+  const codeBlock = page.getBlockById(id) as CodeBlockModel;
+
+  models.forEach(model => {
+    if (model.text instanceof Text) {
```
Comment (lawvs) [inline packages/blocks/src/page-block/utils/container-operations.ts:90]: Maybe we need further discussion about just skipping other blocks...

**Review by lawvs — COMMENTED** [review]
Thanks for your hard work.

**Comment by himself65** [comment]
I saw a regression in https://github.com/toeverything/blocksuite/issues/378

**Review by himself65 — CHANGES_REQUESTED** [review]
image
image

css is incorrect.

File: packages/blocks/src/page-block/utils/container-operations.ts (hunk 8 above)
Comment (thorseraq) [reply packages/blocks/src/page-block/utils/container-operations.ts:90]: OK! Not converting some type of blocks also make sense

File: packages/blocks/src/components/format-quick-bar/format-bar-node.ts [hunk 9]
```diff
@@ -12,10 +12,13 @@ import {
 } from '../../__internal__/utils/index.js';
 import { toast } from '../toast.js';
 import './button';
-import { formatButtons, paragraphButtons } from './config.js';
+import {
+  formatButtons,
+  noneCodeBlockSelected,
```
[the remark this answers is not in the document]
Comment (thorseraq) [reply packages/blocks/src/components/format-quick-bar/format-bar-node.ts:17]: OK! Expecting a discussion :) There should be a more decent way

File: packages/blocks/src/components/format-quick-bar/icons.ts (hunk 5 above)
Comment (thorseraq) [reply packages/blocks/src/components/format-quick-bar/icons.ts]: > image
> 
> image
> 
> css is incorrect.

Thanks! It was caused by wrong z-index, and has been fixed

**Comment by thorseraq** [comment]
Thanks! It was caused by wrong z-index, and has been fixed
cc @Himself65

**Comment by doodlewind** [comment]
I thought the list style regression has been fixed, are we good to merge @lawvs @Himself65?

**Review by lawvs — APPROVED** [review]
LGTM!

**Review by doodlewind — APPROVED** [review]
LGTM 😀

**Pull request merged**